src/protocol/requests/fetch/v4/response.js

```diff
@@ -26,7 +26,13 @@ const MAGIC_OFFSET = 16
  */
 
 const decodeMessages = async decoder => {
-  const messagesBuffer = decoder.readBytes()
+  const messagesSize = decoder.readInt32()
+
+  if (messagesSize <= 0 || !decoder.canReadBytes(messagesSize)) {
```

Comment:
I get what this is doing, but it feels really strange that we'd get `messageSize` > 0 and yet not be able to read enough bytes, and to then return an empty set of messages. Feels wrong. In that situation, shouldn't we buffer until we have enough bytes to read the expected `messageSize`?

Reply:
Discussed outside of Github. Basically the response is complete, so maybe the broker is trying to tell us something by returning a partial record batch, but either way, by returning an empty set of messages here, we should get the complete record batch on the next fetch.

Related: https://github.com/dpkp/kafka-python/blob/ed85c58726f867d956f705194765b18ed3ac3855/kafka/protocol/message.py#L173